connection.ts

```diff
@@ -0,0 +1,195 @@
+import { BufReader, BufWriter } from "./vendor/https/deno.land/std/io/bufio.ts";
+import {
+  CommandExecutor,
+  CommandFunc,
+  RedisRawReply,
+  muxExecutor,
+  sendCommand,
+} from "./io.ts";
+
+// type Reader = Deno.Reader;
+// type Writer = Deno.Writer;
+type Closer = Deno.Closer;
+
+export type RedisConnectOptions = {
+  hostname?: string;
+  port?: number | string;
+  tls?: boolean;
+  db?: number;
+  password?: string;
+  name?: string;
+  maxRetryCount?: number;
+};
+
+export class RedisConnection {
+  name: string | null = null;
+  closer: Closer | null = null;
+  reader: BufReader | null = null;
+  writer: BufWriter | null = null;
+
+  executor: CommandExecutor | null = null;
+
+  get exec(): CommandFunc<RedisRawReply> {
+    return this.executor!.exec;
+  }
+
+  private _isConnected: boolean = false;
+
+  get isConnected(): boolean {
+    return this._isConnected;
+  }
+
+  private _isClosed: boolean = false;
+
+  get isClosed(): boolean {
+    return this._isClosed;
+  }
+
+  maxRetryCount = 0;
+  private retryCount = 0;
+
+  private connectThunkified: () => Promise<RedisConnection>;
+  private thunkifyConnect(
+    hostname: string,
+    port: string | number,
+    options: RedisConnectOptions,
+  ): () => Promise<RedisConnection> {
+    return async () => {
+      const dialOpts: Deno.ConnectOptions = {
+        hostname,
+        port: parsePortLike(port),
+      };
+      if (!Number.isSafeInteger(dialOpts.port)) {
+        throw new Error("deno-redis: opts.port is invalid");
+      }
+      const conn: Deno.Conn = options?.tls
+        ? await Deno.connectTls(dialOpts)
+        : await Deno.connect(dialOpts);
+
+      if (options.name) this.name = options.name;
+      if (options.maxRetryCount) this.maxRetryCount = options.maxRetryCount;
+
+      this.closer = conn;
+      this.reader = new BufReader(conn);
+      this.writer = new BufWriter(conn);
+      this.executor = muxExecutor(this, this.maxRetryCount > 0);
+
+      this._isClosed = false;
+      this._isConnected = true;
+
+      try {
+        if (!!options?.password) await this.authenticate(options.password);
```

Comment:
@sebastienfilion How about this change?
```suggestion
        if (options?.password != null) await this.authenticate(options.password);
```

Reply:
Uh. You're right... Thank you for all your support. I am sorry about the back and forth, I could have been more careful.